File: ReactNative/src/modules/todo/services/get-todos.service.ts
import { ITodos } from '../../../@types/todo';
import { TokenAxios } from '../../common/utils/token-axios';
import { TODO_URL } from '../../common/utils/url';

export async function requestGetTodos(): Promise<ITodos> {
  const AXIOS = await TokenAxios();
  return await AXIOS.get(TODO_URL);
}

File: flutter_todo/lib/src/modules/todo/services/get_todos_service.dart
import 'package:flutter_todo/src/modules/common/utils/token_dio.dart';
import 'package:flutter_todo/src/modules/common/utils/url.dart';
import 'package:flutter_todo/src/modules/todo/models/todo_model.dart';

Future<List<Todo>> requestGetTodos() async => (await tokenDio())
    .get(todoUrl)
    .then((response) => (response.data['todos'] as List<dynamic>)
        .map((json) => Todo.fromJson(json))
        .toList())
    .catchError((error) => throw error);
